Assignment2/StockServiceFactory.java

```diff
@@ -0,0 +1,26 @@
+package assignment2;
+
+/**
+ * Returns a concrete implementations of the interface
+ * 
+ * @author Alyson Hudak
+ */
+
+public class StockServiceFactory 
+{
+    
+    /*Based on the string entered by user, returns the appropriate StockQuote */
```

Comment:
this is not quite right. 
first, never return null unless your code absolutely has to. Typically this can be true when querying a database but more other times it can be avoid. 
See https://en.wikipedia.org/wiki/Tony_Hoare see billion-dollar mistake
second, the factory should just return a BasicStockService and the method to get the service should not take any parameters.